### PR TITLE
fix: update color picker property control interaction

### LINKED ISSUE #4413: [Bug] Color picker covers the input field in small property panes
## Description
Open the color picker inside the container widget, see that the input field is covered. 
Screen Shot 2021-05-11 at 2 51 57 AM

### LINKED ISSUE #6978: [Bug] Color Picker doesn't work
## Description

Color picker not working for buttons, icon button, menu button, and container.
Not sure if there are others

### Steps to reproduce the behavior:
[LOOM DEMO](https://www.loom.com/share/35100b1c91f04a01b9a93ca4dbabaf33)

### Important Details

- Version: [Cloud / Self-Hosted vx.x]
- OS: [e.g.MacOSX]
- Browser [e.g. chrome,]
- Environment [release]

### PR DESCRIPTION
**Pull request by vibhandikyash**
## Description

> The Color picker popover was covering up the textbox. Fixed by changing the popover target on click of the color palette.

Fixes #6978  
~~Fixes #4413~~

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Tested locally.
- Added Test cases.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: FIX/4413-color-picker-interaction-update 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.94 **(0.02)** | 36.83 **(0)** | 33.66 **(0.02)** | 55.48 **(0.01)**
 :green_circle: | app/client/src/components/ads/ColorPickerComponent.tsx | 37.84 **(9.27)** | 63.64 **(5.31)** | 0 **(0)** | 40 **(8.18)**</details>

### CODE REVIEW

**Review by riodeuno — CHANGES_REQUESTED** [review]
Code looks fine to me. Please resolve the merge conflicts.

**Review by somangshu — DISMISSED** [review]
Approving to run tests after conflicts are resolved

**Comment by somangshu** [comment]
The Deploy preview LGTM, The only change I see is that while the inputs are navigated based on the tab key, the colour poper does not come up, We have just fixed this somewhere else

**Comment by somangshu** [comment]
@vibhandikyash please confirm if this failing test case is related

**Comment by vibhandikyash** [comment]
@somangshu No, this is unrelated to my changes.

**Comment by somangshu** [comment]
@vibhandikyash please clear this conflict and merge the latest release here

File: app/client/src/components/ads/ColorPickerComponent.tsx [hunk 1]
```diff
@@ -151,7 +151,6 @@ interface ColorPickerProps {
 
 function ColorPickerComponent(props: ColorPickerProps) {
   const [color, setColor] = React.useState(props.color);
-  const [isOpen, setOpen] = React.useState(false);
```
Comment (somangshu) [inline app/client/src/components/ads/ColorPickerComponent.tsx:154]: @vibhandikyash What about the rest of the changes added with this commit?

**Review by somangshu — APPROVED** [review]
Approving to run tests

File: app/client/src/components/ads/ColorPickerComponent.tsx (hunk 1 above)
Comment (vibhandikyash) [reply app/client/src/components/ads/ColorPickerComponent.tsx:154]: @somangshu  Removed this as there was some code related to picker open and close on tab.

**Comment by somangshu** [comment]
@vibhandikyash we only want to fix #6978 now not go ahead with this behaviour

**Review by riodeuno — APPROVED** [review]
Approving for tests

**Comment by somangshu** [comment]
@vibhandikyash we only need to fix #6978 and revert the current changes in this PR

**Pull request closed without merging**